// Demo of error handling in Zex
define error {
    let message = "unknown error";

    fn construct(message) {
        this.message = message;
    }

    fn error() {
        return this.message;
    }
}

let test = new error("Failed to open file.") @catch err;
if err {
    printerr(err);
}
